### ISSUE TITLE
Module 7 Ex 2 Task 3 Step 17

### ISSUE THREAD

**Issue by XtremeLabsSupport** [MicrosoftLearning/20777A---Implementing-Microsoft-Azure-Cosmos-DB-Solutions#7]
public static void Run(IReadOnlyList<Document> input, ILogger log) gets changed to
public static async Task Run(IReadOnlyList<Document> documents, TraceWriter log)
It does this without changing the following if statement if (input != null && input.Count > 0), I am relatively sure one of the statements will have to change its variable name as I was getting errors about this when viewing the log. Also I am getting a metadata issue when I try to add the Microsoft.ServiceBus reference. This was an issue last year for a time after a push however it was later resolved. https://social.msdn.microsoft.com/Forums/en-US/4bbd43b1-6488-464e-9a1c-7ce00e2a65a6/compilation-error-referencing-microsoftservicebus?forum=AzureFunctions.

While running the function in Ex 4 Task 6 Step 14 I received errors involving the functions rather than the namespaces, ILogger seems to have been deprecated according to the log files.

**Comment by JohnPWSharp** [comment]
Originally, the template code generated by the function wizard used TraceWriter, but this has since changed to ILogger. I have updated the lab instructions accordingly, and corrected the name of the variable (it should be **input**)

Concerning the issue with the ServiceBus library and  Azure Function Apps, the general wisdom seems to be "wait and things will resolve themselves once various libraries and tools have been synced in Azure". Therefore, I will park this issue for a couple of weeks. If it has not *self-resolved* by the end of the month, I will come back to it. 

In Ex4, I have changed log.Info (the original TraceWriter function) to log.LogInformation (the function provided by ILogger)
